src/timer.rs

```diff
@@ -58,12 +59,74 @@ pub mod config {
         fn default() -> Self {
             Self {
                 divider: 80,
-                #[cfg(any(esp32s2, esp32s3, esp32c3))]
+                #[cfg(not(esp32))]
                 xtal: false,
                 auto_reload: false,
             }
         }
     }
+
+    #[cfg(not(esp_idf_version_major = "4"))]
+    #[allow(clippy::upper_case_acronyms)]
+    pub(crate) enum ClockSource {
```

Comment:
Nit: why `pub(crate)`?

Reply:
I meant why is it with a crate visibility and not **private**, but then again - just a nit.